src/Lecture2.hs

```diff
@@ -261,7 +422,14 @@ data EvalError
 It returns either a successful evaluation result or an error.
 -}
 eval :: Variables -> Expr -> Either EvalError Int
-eval = error "TODO"
+eval _ (Lit value)    = Right value
+eval vars (Var var)   = case lookup var vars of
+                          Just value -> Right value
+                          Nothing    -> Left $ VariableNotFound var
+eval vars (Add e1 e2) = case (eval vars e1, eval vars e2) of
+                          (Right value1, Right value2) -> Right $ value1 + value2
+                          (Left e, _)                  -> Left e
+                          (_, Left e)                  -> Left e
```

Comment:
Perfect 🆗 
Once you learn about monads, you'll be able to simplify this case to a single line 👍🏻